StortSpelProjekt/Engine/src/Physics/Physics.cpp

```diff
@@ -28,6 +28,22 @@ bool Physics::CheckOBBCollision(const DirectX::BoundingOrientedBox* obb1, const
 	return obb1->Intersects(*obb2);
 }
 
+std::vector<Entity*> Physics::SpecificCollisionCheck(const DirectX::BoundingOrientedBox* obb1)
+{
+	std::vector<Entity*> entityHitList;
+	for (int i = 0; i < m_CollisionEntities.size(); i++)
+	{
+		if (obb1 != m_CollisionEntities.at(i)->GetComponent<component::BoundingBoxComponent>()->GetOBB())
```

Comment:
You could probably use a std::map here to avoid looping through all collision entities.